src/manifold/src/manifold.cpp

```diff
@@ -845,4 +848,141 @@ Manifold Manifold::Hull() const { return Hull(GetMesh().vertPos); }
 Manifold Manifold::Hull(const std::vector<Manifold>& manifolds) {
   return Compose(manifolds).Hull();
 }
+
+// TODO: Handle Joggling in the Fracture Function Directly?
+/**
+ * Compute the voronoi fracturing of this Manifold into convex chunks.
+ *
+ * @param pts A vector of points over which to fracture the manifold.
+ * @param pts A vector of weights controlling the relative size of each chunk.
+ */
+std::vector<Manifold> Manifold::Fracture(
+    const std::vector<glm::dvec3>& pts,
+    const std::vector<double>& weights) const {
+  std::vector<Manifold> output;
+  output.reserve(pts.size());
+
+  Box bounds = BoundingBox();
+  glm::vec3 min = bounds.min - 0.1f;
+  glm::vec3 max = bounds.max + 0.1f;
+  double V = (max.x - min.x) * (max.y - min.y) * (max.z - min.z);
+  double Nthird = powf((double)pts.size() / V, 1.0f / 3.0f);
+  voro::container_poly container(min.x, max.x, min.y, max.y, min.z, max.z,
+                                 std::round(Nthird * (max.x - min.x)),
+                                 std::round(Nthird * (max.y - min.y)),
+                                 std::round(Nthird * (max.z - min.z)), false,
+                                 false, false, pts.size());
+
+  bool hasWeights = weights.size() == pts.size();
+  for (size_t i = 0; i < pts.size(); i++) {
+    container.put(i, pts[i].x, pts[i].y, pts[i].z,
+                  hasWeights ? weights[i] : 1.0f);
+  }
+  voro::voronoicell c(container);
+  voro::c_loop_all vl(container);
+  if (vl.start()) do {  // TODO: Parallelize this loop!
+      if (container.compute_cell(c, vl)) {
+        std::vector<glm::vec3> verts;
+        verts.reserve(c.p);
+        int id;
+        double x, y, z, r;
+        vl.pos(id, x, y, z, r);
+        for (size_t i = 0; i < c.p; i++) {
+          verts.push_back(glm::vec3(x + 0.5 * c.pts[(vl.ps * i) + 0],
+                                    y + 0.5 * c.pts[(vl.ps * i) + 1],
+                                    z + 0.5 * c.pts[(vl.ps * i) + 2]));
+        }
+        Manifold outputManifold =
+            Hull(verts) ^
+            *this;  // TODO: Replace this intersection with a voro++ wall
+                    // implementation or cutting the cell directly...
+        if (outputManifold.GetProperties().volume > 0.0) {
+          output.push_back(outputManifold);
+        }
+        verts.clear();
+      }
+    } while (vl.inc());
+  return output;
+}
+std::vector<Manifold> Manifold::Fracture(
+    const std::vector<glm::vec3>& pts,
+    const std::vector<float>& weights) const {
```

Comment:
I'd like to avoid having two APIs for different precisions.

Reply:
Is it okay to expect everything to converge to `double` only with #659 ?

Reply:
IMO yes.

Reply:
yes